application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -252,9 +256,11 @@ public void upgradeRuleNodes() {
                                 ruleNode.setConfiguration(upgradeRuleNodeConfigurationResult.getSecond());
                             }
                             ruleNode.setConfigurationVersion(toVersion);
-                            ruleChainService.saveRuleNode(TenantId.SYS_TENANT_ID, ruleNode);
-                            log.debug("Successfully upgrade rule node with id: {} type: {} fromVersion: {} toVersion: {}",
-                                    ruleNodeId, ruleNodeTypeForLogs, fromVersion, toVersion);
+                            jpaExecutorService.submit(() -> {
```

Comment:
This is a very naive code since you do not wait for the task to be finished. You should collect listenable futures and wait for their completion.